src/extractor/guidance/mergable_road_detector.cpp

```diff
@@ -0,0 +1,427 @@
+    graph_walker.TraverseRoad(intersection_node, lhs.eid, left_accumulator, selector);
+    /* if the intersection does not have a right turn, we continue onto the next one once
+     * (skipping over a single small side street)
+     */
+    if (angularDeviation(left_accumulator.intersection.findClosestTurn(90)->angle, 90) >
+        NARROW_TURN_ANGLE)
+    {
+        graph_walker.TraverseRoad(node_based_graph.GetTarget(left_accumulator.via_edge_id),
+                                  left_accumulator.intersection.findClosestTurn(180)->eid,
+                                  left_accumulator,
+                                  selector);
+    }
+    const auto distance_to_triangle = util::coordinate_calculation::haversineDistance(
+        node_coordinates[intersection_node],
+        node_coordinates[node_based_graph.GetTarget(left_accumulator.via_edge_id)]);
+
+    // don't move to far down the road
+    if (distance_to_triangle > 80)
+        return false;
+
+    graph_walker.TraverseRoad(intersection_node, rhs.eid, right_accumulator, selector);
+    if (angularDeviation(right_accumulator.intersection.findClosestTurn(270)->angle, 270) >
+        NARROW_TURN_ANGLE)
+    {
+        graph_walker.TraverseRoad(node_based_graph.GetTarget(right_accumulator.via_edge_id),
+                                  right_accumulator.intersection.findClosestTurn(180)->eid,
+                                  right_accumulator,
+                                  selector);
+    }
+
+    BOOST_ASSERT(!left_accumulator.intersection.empty() && !right_accumulator.intersection.empty());
+
+    // find the closes resembling a right turn
+    const auto connector_turn = left_accumulator.intersection.findClosestTurn(90);
+    /* check if that right turn connects to the right_accumulator intersection (i.e. we have a
+     * triangle)
+     * a connection should be somewhat to the right, when looking at the left side of the
+     * triangle
+     *
+     *    b ..... c
+     *     \     /
+     *      \   /
+     *       \ /
+     *        a
+     *
+     * e.g. here when looking at `a,b`, a narrow triangle should offer a turn to the right, when
+     * we want to connect to c
+     */
+    if (angularDeviation(connector_turn->angle, 90) > NARROW_TURN_ANGLE)
+        return false;
+
+    const auto num_lanes = [this](const MergableRoadData &road) {
+        return std::max<std::uint8_t>(
+            node_based_graph.GetEdgeData(road.eid).road_classification.GetNumberOfLanes(), 1);
+    };
+
+    // the width we can bridge at the intersection
+    const auto assumed_road_width = (num_lanes(lhs) + num_lanes(rhs)) * ASSUMED_LANE_WIDTH;
+    const auto distance_between_triangle_corners = util::coordinate_calculation::haversineDistance(
+        node_coordinates[node_based_graph.GetTarget(left_accumulator.via_edge_id)],
+        node_coordinates[node_based_graph.GetTarget(right_accumulator.via_edge_id)]);
+    if (distance_between_triangle_corners > (assumed_road_width + 10))
```

Comment:
Also feel free to create more variables, I feel like a lot of your function calls are endlessly long since you subscript, call functions etc.